### DISCUSSION TITLE
[BUG]ERROR 2002 (HY000): Can’t connect to local MySQL server through socket ‘/var/run/mysqld/mysqld.sock’ (2)

### DISCUSSION THREAD

**Discussion by dengxiwang** [1Panel-dev/1Panel#6941] [Q&A]
### 联系方式

dengxiwang@aliyun.com

### 1Panel 版本

v1.10.20-lts

### 问题描述

我用的是绿联云nas，debian12的系统底层，将1panel安装在了/volume1/server目录下，在应用商店安装mysql应用，安装是正常的，但是点击菜单中的数据库，点击设置，会显示ERROR 2002 (HY000): Can’t connect to local MySQL server through socket ‘/var/run/mysqld/mysqld.sock’ (2)，如果安装别的应用使用这个数据库，也是提示这样的错误。
![1730709456032](https://github.com/user-attachments/assets/bf2a3fa6-1980-4815-9ae4-cbd604521da4)
但是如果在应用商店中安装5.7版本的数据库，是正常的，不会报错，其他应用也可以正常使用，但是5.7版本的话应用商店的升级不会显示升级到8.x的版本
容器日志文件：
[_volume1_server_1panel_apps_mysql_mysql_docker-compose.yml-20241104163523.log](https://github.com/user-attachments/files/17616004/_volume1_server_1panel_apps_mysql_mysql_docker-compose.yml-20241104163523.log)

### 重现步骤

将1panel安装在了/volume1/server目录下，在应用商店安装mysql应用，安装是正常的，但是点击菜单中的数据库，点击设置，会显示ERROR 2002 (HY000): Can’t connect to local MySQL server through socket ‘/var/run/mysqld/mysqld.sock’ (2)

### 期待的正确结果

一切正常

**Comment by wanghe-fit2cloud** [comment]
这可能与 NAS 环境有关。建议先排查 MySQL 容器日志，然后检查 `/var/run/mysqld/` 目录权限是否正确等。
